SonarQube.Common/AnalysisConfig/AnalysisSetting.cs

```diff
@@ -48,6 +48,20 @@ public class AnalysisSetting
         /// </remarks>
         private static readonly Regex ValidSettingKeyRegEx = new Regex(@"^\w[\w\d\.-]*$", RegexOptions.Compiled);
 
+        //TODO: this expression only works for single-line values
```

Comment:
Moved from the FilePropertyProvider so it can be shared.
I've changed regex so that the value cannot be empty. Previously it allowed "key=". @dbolkensteyn should we allow empty values?